From 45afbe9f7355902d30ae8e4a45843421ee1f1b05 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Thu, 11 Jul 2013 15:13:32 +0900
Subject: [PATCH 2/3] dfb-video-example: clear sigaction structure instance

This fix clears unused structure members of sigaction to avoid
the malfunction of signal processing.
---
 ext/directfb/dfb-video-example.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/directfb/dfb-video-example.c b/ext/directfb/dfb-video-example.c
index a9b1d8b..aaa6e7c 100644
--- a/ext/directfb/dfb-video-example.c
+++ b/ext/directfb/dfb-video-example.c
@@ -604,6 +604,7 @@ main (int argc, char *argv[])
   g_signal_connect (demuxer, "pad-added", G_CALLBACK (on_pad_added), sink);
 
   /* set signal handler */
+  memset (&action, 0, sizeof (action));
   action.sa_handler = play_handler;
   sigaction (SIGQUIT, &action, NULL);
   sigaction (SIGINT, &action, NULL);
-- 
1.8.1.2

